### ISSUE TITLE
Response with different protobuf type causes assertion error, not error callback

### ISSUE THREAD

**Issue by evanj** [grpc/grpc-web#419]
The gRPC server response could be anything, including a different protocol buffer message than the one the client expects. In this case, the decoding will throw an assertion error, and the error callback is never called. It appears that `jspb.BinaryReader`'s API is that is throws these assert errors on parse failures. This means grpc-web needs to catch them. The NodeJS implementation reports this case as an error with `.code = grpc.status.INTERNAL` and `.details = 'Failed to parse server response'`

The stack trace looks like:

```
AssertionError: Assertion failed
    at new goog.asserts.AssertionError (node_modules/google-protobuf/google-protobuf.js:98:603)
    at Object.goog.asserts.doAssertFailure_ (node_modules/google-protobuf/google-protobuf.js:99:126)
    at Object.goog.asserts.assert (node_modules/google-protobuf/google-protobuf.js:99:385)
    at jspb.BinaryReader.readString (node_modules/google-protobuf/google-protobuf.js:357:64)
    at Function.proto.interop.EchoResponse.deserializeBinaryFromReader (interop_pb.js:285:49)
    at module$contents$grpc$web$GrpcWebClientReadableStream_GrpcWebClientReadableStream.proto.interop.EchoResponse.deserializeBinary [as responseDeserializeFn_] (interop_pb.js:266:37)
    at goog.net.XhrIo.<anonymous> (node_modules/grpc-web/index.js:433:153)
    at goog.net.XhrIo.goog.events.EventTarget.fireListeners (node_modules/grpc-web/index.js:281:298)
    at Function.goog.events.EventTarget.dispatchEventInternal_ (node_modules/grpc-web/index.js:283:477)
    at goog.net.XhrIo.goog.events.EventTarget.dispatchEvent (node_modules/grpc-web/index.js:278:298)
```

This is probably one cause of #415, but at least this one logs something.

**Comment by guyguy333** [comment]
Having the same issue with grpc-web v1.1.0. In our case, the error happens when proto is different between client and server. Unfortunately, we have no solution to catch the error (which is annoying). We just have this log: `uncaught exception: AssertionError: Assertion failed` in console. Here is stack: 

```
./src/proto/core/common_pb.js/proto.core.Product.deserializeBinaryFromReader
src/proto/core/common_pb.js:2561
./node_modules/google-protobuf/google-protobuf.js/</goog.debug.Error
node_modules/google-protobuf/google-protobuf.js:81
./node_modules/google-protobuf/google-protobuf.js/</goog.asserts.AssertionError
node_modules/google-protobuf/google-protobuf.js:81
./node_modules/google-protobuf/google-protobuf.js/</goog.asserts.doAssertFailure_
node_modules/google-protobuf/google-protobuf.js:82
./node_modules/google-protobuf/google-protobuf.js/</goog.asserts.assert
node_modules/google-protobuf/google-protobuf.js:83
./node_modules/google-protobuf/google-protobuf.js/</jspb.BinaryDecoder.prototype.advance
node_modules/google-protobuf/google-protobuf.js:354
./node_modules/google-protobuf/google-protobuf.js/</jspb.BinaryReader.prototype.skipDelimitedField
node_modules/google-protobuf/google-protobuf.js:377
./node_modules/google-protobuf/google-protobuf.js/</jspb.BinaryReader.prototype.skipField
node_modules/google-protobuf/google-protobuf.js:380
```

@stanley-cheung Any hints to solve this issue ?

Thanks

**Comment by elstr** [comment]
Hi! Is there any workaround for this?

**Comment by stanley-cheung** [comment]
This should be fixed in #903.
